src/JSONManager.py

```diff
@@ -0,0 +1,68 @@
+import src.utils as utils
+import src.errors as errors
+import os
+
+
+class JSONManager(metaclass=utils.SingletonMeta):
+    def __init__(self):
+        self._file_id_counter = 1
+        self._open_jsons = {}
+        self._open_datasets = {}
+
+    def open_json(self, path: str) -> int:
```

Comment:
as we already mension that this is a json manager, I think we should not use word "json" in method names.

Reply:
I renamed the methods.